### ISSUE TITLE
unclean reactor error from testPortforward on glib2 reactor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2867]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2867|
|Type|defect|
|Created|2007-11-05 02:34:19Z|

===============================================================================
[ERROR]: twisted.test.test_protocols.Portforwarding.testPortforward

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
Selectables:
<<class 'twisted.internet.tcp.Client'> to ('127.0.0.1', 47104) at 2aaaae59e050>
-------------------------------------------------------------------------------

<details><summary>Searchable metadata</summary>

```
trac-id__2867 2867
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1194230059000000 1194230059000000
changetime__1194290305000000 1194290305000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [21578]) Fix the unclean error

Refs [#2867](https://github.com/twisted/twisted/issues/2867)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve removed owner|
|-|-|

Ready to review in **test-portforward-unclean-2867**.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @therve|
|-|-|

Ick.  The existing try/except in Portforwarding.tearDown is not very nice.  The new ones also aren't very nice.  I guess this is for AttributeError or something?  Maybe it could at least specifically catch that.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

(In [21583]) Use specfic exception.

Refs [#2867](https://github.com/twisted/twisted/issues/2867)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @therve|
|-|-|

+1

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set status to `closed`|
|-|-|

(In [21586]) Merge test-portforward-unclean-2867

Author: therve
Reviewer: exarkun
Fixes [#2867](https://github.com/twisted/twisted/issues/2867)

Fix an unclean error in testPortforward with glib2 reactor.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
